src/functions-reference/higher-order_functions.Rmd

```diff
@@ -412,7 +412,129 @@ the same size:
+Because there can be any number of these arguments and they can be of different types,
+they are denoted below as `...`, and the types of these arguments,
+also represented by `...` in the DAE solver call, must match the types of the arguments represented by
+`...` in the user-supplied system function.
+
+### The DAE solver
+
+<!-- array[] vector; dae; (function dae, vector initial_state, vector initial_state_derivative, real initial_time, array[] real times, ...); -->
+\index{{\tt \bfseries dae }!{\tt (function dae, vector initial\_state, vector initial\_state\_derivative, real initial\_time, array[] real times, ...): array[] vector}|hyperpage}
+
+`array[] vector` **`dae`**`(function dae, vector initial_state, vector initial_state_derivative, real initial_time, array[] real times, ...)`<br>\newline
+Solves the DAE system using the backward differentiation formula (BDF)
+method [@serban_user:2021].
+`r since("2.29")`
+
+<!-- array[] vector; dae_tol; (function dae, vector initial_state, vector initial_state_derivative, real initial_time, array[] real times, data real rel_tol, data real abs_tol, int max_num_steps, ...); -->
+\index{{\tt \bfseries dae\_tol }!{\tt (function dae, vector initial\_state, vector initial\_state\_derivative, real initial\_time, array[] real times, data real rel\_tol, data real abs\_tol, int max\_num\_steps, ...): array[] vector}|hyperpage}
+
+`array[] vector` **`dae_tol`**`(function dae, vector initial_state, vector initial_state_derivative, real initial_time, array[] real times, data real rel_tol, data real abs_tol, int max_num_steps, ...)`<br>\newline
+Solves the DAE system for the times provided using the backward differentiation formula (BDF) method with additional control
+parameters for the solver.
+`r since("2.29")`
+
+### DAE system function
+
+The first argument to the DAE solver is the DAE residual
+function. The DAE residual function must have a `vector` return type, and the
+first three arguments must be a `real`, `vector`, and `vector`, in that order. These three
+arguments are followed by the variadic arguments that are passed through from
+the DAE solver function call:
+
+```stan
+  vector dae(real time, vector state, vector state_derivative, ...)
+```
+
+The DAE residual function should return the residuals the time and state provided. The length of the returned
+vector must match the length of the state input into the function.
+
+The arguments to this function are:
+
+*   *`time`*, the time to evaluate the DAE system
+
+*   *`state`*, the state of the DAE system at the time specified
+
+*   *`state_derivative`*, the time derivatives of the state of the DAE system at the time specified
+
+*   *`...`*, sequence of arguments passed unmodified from the DAE solve
+function call. The types here must match the types in the `...` arguments of the
+DAE solve function call.
+
+### Arguments to the DAE solver {#dae-solver}
+
+The arguments to the DAE solver are
+
+*   *`ode`*: DAE residual function,
```

Comment:
`ode` is wrong here, should match above (see comment about suggested name change for these placeholders)